starter-code/public/javascripts/financial-data.js

```diff
@@ -0,0 +1,64 @@
+const coinDeskApi = axios.create({
+    baseURL: 'http://api.coindesk.com/v1/bpi/historical/close.json'
+})
+
+
+function getCoindeskData() {
+    return coinDeskApi.get()
+        .then(response => {
+            return response.data.bpi; // <---- that data
+        })
+        .catch(err => {
+            console.error(err)
+        });
+}
+
+
+function buildChart(data) { 
+    var ctx = document.getElementById("myChart");
+    var myChart = new Chart(ctx, {
+        type: 'bar',
+        data: {
+            labels: ["Red", "Blue", "Yellow", "Green", "Purple", "Orange"],
+            datasets: [{
+                label: '# of Votes',
+                data: [12, 19, 3, 5, 2, 3],
+                backgroundColor: [
+                    'rgba(255, 99, 132, 0.2)',
+                    'rgba(54, 162, 235, 0.2)',
+                    'rgba(255, 206, 86, 0.2)',
+                    'rgba(75, 192, 192, 0.2)',
+                    'rgba(153, 102, 255, 0.2)',
+                    'rgba(255, 159, 64, 0.2)'
+                ],
+                borderColor: [
+                    'rgba(255,99,132,1)',
+                    'rgba(54, 162, 235, 1)',
+                    'rgba(255, 206, 86, 1)',
+                    'rgba(75, 192, 192, 1)',
+                    'rgba(153, 102, 255, 1)',
+                    'rgba(255, 159, 64, 1)'
+                ],
+                borderWidth: 1
+            }]
+        },
+        options: {
+            scales: {
+                yAxes: [{
+                    ticks: {
+                        beginAtZero:true
+                    }
+                }]
+            }
+        }
+    });
+
+
+document.getElementById("coindesk-button").onclick = function(){
```

Comment:
Nothing happens when you click the button because this function is running before the button exists. You need to wrap this with window.onload.